data/src/test/java/org/apache/iceberg/data/TestLocalScan.java

```diff
@@ -554,8 +552,7 @@ public void testFilterWithDateAndTimestamp() throws IOException {
             required(3, "date", Types.DateType.get()),
             required(4, "time", Types.TimeType.get()));
 
-    File tableLocation = Files.createTempDirectory(tempDir.toPath(), "junit").toFile();
-    assertThat(tableLocation.delete()).isTrue();
+    File tableLocation = new File(tempDir, "junit");
```

Comment:
all of these need some random part (either UUID or system.nanotime) as in the previous version `junit` was only the prefix of the generated path

Reply:
Done